js/index.js

```diff
@@ -711,11 +420,13 @@ $(function() {
                     setupEditor();
                 }).fail(function() {
                     console.log("Unable to load gist contents");
+                    $('#code_textarea').val("-- Unable to load gist contents. Gist = " + id);
```

Comment:
Solution I see without restructuring things too much is to parameterize

```
var loadOptions = function(tryLoad) { ...

var setupEditor = function(tryLoad) { ...
```

Then this logic only needs to change:
https://github.com/rintcius/trypurescript/blob/dcb8522982f2ca0fd22e3f7a2177dec271ef244e/js/index.js#L88-L90

into 

```
      if (tryLoad && $('#code_textarea').val() === "") {
          loadFromGist(backend.mainGist);
      }
```

I think splitting up would mean taking the `if` statement above out of `loadOptions` and into its own function but I think this would end up to be more complex than parameterizing (because AFAICS the only change we'd want in the flow is this `if` statement)